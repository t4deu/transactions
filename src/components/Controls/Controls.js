// @flow
import React, { Component } from 'react';
import { ActionButton } from 'react-native-material-ui';

type Props = {
  onAddCredit: Function,
  onAddDebit: Function,
};

export default class Controls extends Component<Props> {
  onPress = (key: string) => {
    if (key === 'add') {
      this.props.onAddCredit();
    } else if (key === 'remove') {
      this.props.onAddDebit();
    }
  };

  render() {
    return (
      <ActionButton
        transition="toolbar"
        onPress={this.onPress}
        actions={[
          {
            icon: 'remove',
            label: 'Débito',
            name: 'remove',
          },
          {
            icon: 'add',
            label: 'Crédito',
            name: 'add',
          },
        ]}
      />
    );
  }
}
