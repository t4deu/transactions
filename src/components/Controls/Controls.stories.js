import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { View } from 'react-native';

import Controls from './Controls';

storiesOf('Controls', module)
  .addDecorator(story => <View style={{ flex: 1, marginTop: 200 }}>{story()}</View>)
  .add('Actions', () => <Controls onAddCredit={() => {}} onAddDebit={() => {}} />);
