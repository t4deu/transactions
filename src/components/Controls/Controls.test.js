import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Controls from './Controls';

describe('<Controls>', () => {
  it('renders without issues', () => {
    const component = shallow(<Controls title="CrÃ©dito" onAddCredit={() => {}} onAddDebit={() => {}} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
