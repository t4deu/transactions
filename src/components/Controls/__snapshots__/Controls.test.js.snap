// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls> renders without issues 1`] = `
<ActionButton
  actions={
    Array [
      Object {
        "icon": "remove",
        "label": "Débito",
        "name": "remove",
      },
      Object {
        "icon": "add",
        "label": "Crédito",
        "name": "add",
      },
    ]
  }
  hidden={false}
  icon="add"
  onLongPress={null}
  onPress={[Function]}
  rippleColor={null}
  style={Object {}}
  transition="toolbar"
/>
`;
